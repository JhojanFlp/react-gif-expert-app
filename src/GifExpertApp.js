import React, { useState } from 'react';

import AddCategory from './components/AddCategory';
import Grid from './components/Grid';

const GifExpertApp = ({ defaultCategories = []}) => {
  const [categories, setCategories] = useState(defaultCategories);

  const addCategory = (category = '') => {
    if (!category) return
    setCategories([ category, ...categories ]);
  }

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory addCategory={addCategory} />
      <hr />
      <ol>
        {
          categories.map(category => 
            <Grid 
              key={category}
              category={category}
            />
          )
        }
      </ol>
    </>
  )
}

export default GifExpertApp
