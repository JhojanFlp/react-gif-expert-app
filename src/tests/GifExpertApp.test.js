import React from 'react';
import { shallow } from 'enzyme';

import GifExpertApp from '../GifExpertApp.js';

describe('Test in GifExpertApp', () => {
  const categories = ['Kimetsu', 'Dragon Ball'];
  test('should render component', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  })
  test('should categories', () => {
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Grid').length).toBe(categories.length)

  })
  
})
