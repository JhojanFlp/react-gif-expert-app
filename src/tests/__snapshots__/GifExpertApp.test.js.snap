// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifExpertApp should categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    addCategory={[Function]}
  />
  <hr />
  <ol>
    <Grid
      category="Kimetsu"
      key="Kimetsu"
    />
    <Grid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Test in GifExpertApp should render component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    addCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
